agent/internal/command.go

```diff
@@ -0,0 +1,80 @@
+package internal
+
+import (
+	"github.com/sqreen/go-agent/agent/internal/backend/api"
+	"github.com/sqreen/go-agent/agent/internal/plog"
+)
+
+type CommandManager struct {
+	logger   *plog.Logger
+	agent    CommandManagerAgent
+	handlers map[string]CommandHandler
+}
+
+type CommandHandler func() api.CommandResult
+
+// CommandManagerAgent defines the expected agent SDK and allows to easily
+// implement functional tests by mocking it up.
+type CommandManagerAgent interface {
+	InstrumentationEnable() error
+	InstrumentationDisable() error
+}
+
+func NewCommandManager(agent CommandManagerAgent, logger *plog.Logger) *CommandManager {
+	mng := &CommandManager{
+		agent:  agent,
+		logger: plog.NewLogger("command", logger),
+	}
+
+	// Note: using Go's reflection to call methods would be slower.
+	mng.handlers = map[string]CommandHandler{
+		"instrumentation_enable": mng.InstrumentationEnable,
+		"instrumentation_remove": mng.InstrumentationRemove,
+	}
+
+	return mng
+}
+
+func (m *CommandManager) Do(commands []api.CommandRequest) map[string]api.CommandResult {
+	if len(commands) == 0 {
+		return nil
+	}
+
+	results := make(map[string]api.CommandResult, len(commands))
+	for _, cmd := range commands {
+		handler, exists := m.handlers[cmd.Name]
+		if !exists {
+			m.logger.Warnf("ignoring unsupported command `%s`", cmd.Name)
```

Comment:
Must return a command failure to BE:
* as a command response 
* as an exception

Reply:
:+1: 
Only command response is fine for now right? Because there is no possible exception here.

Reply:
Technically without the exception report you won't be pro-actively warned about the issue. Link me to a jira ticket for adding exception later and it will ge good for me ;)

Reply:
For exceptions: https://sqreen.atlassian.net/browse/SQR-5752
Fixing the rest of it.

Reply:
Fixed by a95a952. Let me know.